feat: improve `MISSING_EXPORT` warning to suggest `type` modifier

sapphi-red opened the pull request:
There are multiple cases that `MISSING_EXPORT` happens even when TypeScript does not output an error.

1. https://github.com/rolldown/rolldown/pull/4144#issuecomment-2811663851
    - This case only happens when the importer and the importee is both TS. This is because we treat `foo` as a "type" rather than a "value" (since we remove `export default foo` completely. If we treat it as a "value", we should keep that).
2. The case written in https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax (`export { A } from './a'`)
    - This case only happens when the importer and the importee is both TS. This is because if `A` is a "value", `A` would exist after transpilation.

Both of these cases can be fixed by adding the `type` modifier. While setting `verbatimModuleSyntax: true` would make TypeScript to error on these cases, it changes some other behaviors, and is not straightforward to turn that on. This PR improves the error message, so that users can know what to do even if they don't enable `verbatimModuleSyntax`.

refs #4147

sapphi-red commented:
* **#6085** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6085?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6085?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6085).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

IWANABETHATGUY commented:
### Merge activity

* **Sep 9, 5:12 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 9, 5:12 AM UTC**: `IWANABETHATGUY` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 9, 5:46 AM UTC**: The [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) couldn't merge this PR because **it was not satisfying all requirements** (Failed CI: 'node-test (macos-latest) / Node Test').
* **Sep 9, 6:06 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 9, 6:06 AM UTC**: `IWANABETHATGUY` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 9, 6:07 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).

sapphi-red committed:
feat: improve `MISSING_EXPORT` warning to suggest `type` modifier (#6085)

There are multiple cases that `MISSING_EXPORT` happens even when TypeScript does not output an error.

1. https://github.com/rolldown/rolldown/pull/4144#issuecomment-2811663851
    - This case only happens when the importer and the importee is both TS. This is because we treat `foo` as a "type" rather than a "value" (since we remove `export default foo` completely. If we treat it as a "value", we should keep that).
2. The case written in https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax (`export { A } from './a'`)
    - This case only happens when the importer and the importee is both TS. This is because if `A` is a "value", `A` would exist after transpilation.

Both of these cases can be fixed by adding the `type` modifier. While setting `verbatimModuleSyntax: true` would make TypeScript to error on these cases, it changes some other behaviors, and is not straightforward to turn that on. This PR improves the error message, so that users can know what to do even if they don't enable `verbatimModuleSyntax`.

refs #4147
